library/src/scala/NamedTuple.scala

```diff
@@ -19,7 +19,12 @@ object NamedTuple:
 
   def unapply[N <: Tuple, V <: Tuple](x: NamedTuple[N, V]): Some[V] = Some(x)
 
+  @annotation.compileTimeOnly("NamedTupleBuilder.Builder should have been eliminated by calling the build method.")
+  final class NamedTupleBuilder[N <: Tuple]():
+    inline def build[V <: Tuple](x: V): NamedTuple[N, V] = x
+
   extension [V <: Tuple](x: V)
```

Comment:
I believe `withNames` is still useful, as the easiest low-level way in user code to go from a tuple to a named tuple (i.e. not using the named tuple source syntax).

So I don't think it should be deprecated.

Reply:
kept it